syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionTest.java

```diff
@@ -0,0 +1,172 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.syndesis.extensions.core.model;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author lhein
+ */
+public class SyndesisExtensionTest {
+
+	private SyndesisExtension extension;
+
+	@Before
+	public void onStart() throws IOException {
+		File jsonFile = new File("testdata/extdef1.json");
+		this.extension = SyndesisExtension.getJSONFactoryInstance(new FileInputStream(jsonFile));
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getExtensionId()}.
+	 */
+	@Test
+	public void testGetExtensionId() {
+		assertTrue("extensionId value is missing", extension.getExtensionId() != null && extension.getExtensionId().trim().length()>0);
```

Comment:
why checking for emptiness if it is checked for a specific value just after?

I think that
assertThat(extension.getExtensionId()).isEqualTo("my:test") is enough for the test. the message is redundant with the test method name

and same for almost all the tests...